app/src/main/java/com/github/libretube/ui/dialogs/PlayOfflineDialog.kt

```diff
@@ -0,0 +1,44 @@
+package com.github.libretube.ui.dialogs
+
+import android.app.Dialog
+import android.content.DialogInterface
+import android.content.Intent
+import android.os.Bundle
+import androidx.fragment.app.DialogFragment
+import androidx.fragment.app.commit
+import com.github.libretube.R
+import com.github.libretube.constants.IntentData
+import com.github.libretube.databinding.DialogPlayOfflineBinding
+import com.github.libretube.db.obj.DownloadWithItems
+import com.github.libretube.helpers.PreferenceHelper
+import com.github.libretube.ui.activities.MainActivity
+import com.github.libretube.ui.activities.OfflinePlayerActivity
+import com.google.android.material.dialog.MaterialAlertDialogBuilder
+
+class PlayOfflineDialog(
+    private val downloadItem: DownloadWithItems,
+    private val onConfirmResult: ()->Unit,
+    private val onNegativeResult: DialogInterface.OnClickListener
+) : DialogFragment() {
```

Comment:
It's generally better to not pass any data via dialog constructors, as if the activity is recreated the dialog is being recreated without the constructor arguments.

Instead, we should pass the the video information via arguments (`putExtra` and `getExtra`) and use Fragment result listeners as per https://developer.android.com/guide/fragments/communicate#fragment-result.